vignettes/package.Rmd

```diff
@@ -0,0 +1,349 @@
+#>    cargo build --lib --release --manifest-path=./rust/Cargo.toml
+#>        Updating crates.io index
+#>       Compiling proc-macro2 v1.0.27
+#>       Compiling unicode-xid v0.2.2
+#>       Compiling libR-sys v0.2.1
+#>       Compiling syn v1.0.72
+#>       Compiling extendr-engine v0.2.0
+#>       Compiling lazy_static v1.4.0
+#>       Compiling quote v1.0.9
+#>       Compiling extendr-macros v0.2.0
+#>       Compiling extendr-api v0.2.0
+#>       Compiling myextendr v0.1.0 (path/to/myextendr/src/rust)
+#>        Finished release [optimized] target(s) in 19.05s
+#>    gcc -std=gnu99 -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o myextendr.so entrypoint.o -L./rust/target/release #> -lmyextendr -L/usr/lib/R/lib -lR
+#>    installing to /tmp/RtmpfMcL08/devtools_install_e2d6351b843c/00LOCK-myextendr/00new/myextendr/libs
+#>    ** checking absolute paths in shared objects and dynamic libraries
+#> ─  DONE (myextendr)
+#> ✓ Writing 'R/extendr-wrappers.R'.
+#> ℹ Updating myextendr documentation
+#> ℹ Loading myextendr
+#> Writing NAMESPACE
+#> Writing NAMESPACE
+#> Writing hello_world.Rd
+```
+
+You might wonder why compilation is triggered while the function name is just `document()`.
+Well, this is because the compilation is actually needed to generate document
+from Rust code. This is consistent with `devtools::document()`'s behavior for C/C++ codes.
+
+Anyway, by doing above, the following files are updated or generated:
+
+```
+.
+...
+├── NAMESPACE                       ----------(4)
+├── R
+│   └── extendr-wrappers.R          ----------(3)
+├── man
+│   └── hello_world.Rd              ----------(4)
+└── src
+    ├── myextendr.so          ----------(2)
+    └── rust
+        └── target
+            └── release
+                ├── libmyextendr.a   ---(1)
+                ...
+```
+
+1. **`src/rust/target/release/libmyextendr.a`** (the extension depends on the OS): 
+This is the static library built from Rust code. This will be then used for compiling
+shared library `myextendr.so`.
+2. **`src/myextendr.so`** (the extension depends on the OS): This is the
+shared object that is actually called from R.
+3. **`R/extendr-wrappers.R`**: The auto-generated R functions, including roxygen
+comments, goes to this file. The roxygen comments are accordingly converted into
+Rd files and `NAMESPACE`.
+4. **`man/`**, **`NAMESPACE`**: These are generated from roxygen comments.
+
+### Load and use
+
+As all things are done by `rexetndr::document()` already, we can just load it 
+(or install it if you want) and call the function.
+
+``` r
+devtools::load_all(".")
+
+hello_world()
+#> [1] "Hello world!"
+```
+
+## Rust code vs generated R code
+
+While we never edit it by hand, it might be good to know what R code is
+generated from a Rust code. Let's look at `R/extendr-wrappers.R`:
```

Comment:
"from a Rust code" is weird. Maybe "from some Rust code" or "from specific Rust code"?

Reply:
I chose "from the above Rust code." Does this sound fine?

Reply:
Yes, good choice.

Reply:
Thanks.